src/test/regress/expected/propagate_extension_commands.out

```diff
@@ -0,0 +1,166 @@
+-- master add node
+CREATE SCHEMA "extension'test";
+-- use  a schema name with escape character
+SET search_path TO "extension'test";
+-- create an extension on the given search_path
+-- the extension is on contrib, so should be avaliable for the regression tests
+CREATE EXTENSION hstore;
+--  make sure that both the schema and the extension is distributed
+SELECT count(*) FROM citus.pg_dist_object WHERE objid = (SELECT oid FROM pg_extension WHERE extname = 'hstore');
+ count 
+-------
+     1
+(1 row)
+
+SELECT count(*) FROM citus.pg_dist_object WHERE objid = (SELECT oid FROM pg_namespace WHERE nspname = 'extension''test');
+ count 
+-------
+     1
+(1 row)
+
+CREATE TABLE test_table (key int, value hstore);
+SELECT create_distributed_table('test_table', 'key');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+--  make sure that the table is also distributed object now
+SELECT count(*) from pg_dist_partition where logicalrelid='extension''test.test_table'::regclass;
+ count 
+-------
+     1
+(1 row)
+
+CREATE TYPE two_hstores AS (hstore_1 hstore, hstore_2 hstore);
+-- verify that the type that depends on the extension is also marked as distributed
+SELECT count(*) FROM citus.pg_dist_object WHERE objid = (SELECT oid FROM pg_type WHERE typname = 'two_hstores' AND typnamespace = (SELECT oid FROM pg_namespace WHERE nspname = 'extension''test'));
+ count 
+-------
+     1
+(1 row)
+
+-- now CREATE EXTENSION within a transction block and with version defined
+BEGIN;
+	CREATE EXTENSION isn WITH VERSION '1.1' SCHEMA public;
+	-- make sure that the object is distributed 
+	SELECT count(*) FROM citus.pg_dist_object WHERE objid = (SELECT oid FROM pg_extension WHERE extname = 'isn');
+ count 
+-------
+     1
+(1 row)
+
+	-- now, create a reference table relying on the data types
+	CREATE TABLE ref_table (a public.issn);
+	SELECT create_reference_table('ref_table');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+	-- and the extension is created on the workers -- TODO: they return 0 as we are in the transaction block
+	SELECT run_command_on_workers($$SELECT count(*) FROM pg_extension WHERE extname = 'isn'$$);
+ run_command_on_workers 
+------------------------
+ (localhost,57637,t,0)
+ (localhost,57638,t,0)
+(2 rows)
+
+	-- now, update to a newer version
+	ALTER EXTENSION isn UPDATE TO '1.2';
+	-- now change the schema
+	ALTER EXTENSION isn SET SCHEMA "extension'test";
+	-- show that the ALTER EXTENSION commands are propagated
+	SELECT run_command_on_workers($$SELECT extnamespace FROM pg_extension WHERE extname = 'isn'$$);
```

Comment:
Similarly, `run_command_on_workers` doesn't work in transaction block.

Let's move this `SELECT run_command_on_workers($$SELECT extnamespace FROM pg_extension WHERE extname = 'isn'$$);` and the following  `run_command_on_workers ` outside of the transaction block. But, instead of `ROLLBACK `, do a commit, otherwise, we'd not see the results again

Reply:
changes can be seen at 7e956de0c46dc5f986da224d7ff6d3d25ad069c3